apps/src/constants.js

```diff
@@ -26,6 +26,7 @@ exports.TestResults = {
   NO_TESTS_RUN: -1,
 
   // The level was not solved.
+  GENERIC_FAIL: 0,                    // Used by DSL defined levels.
```

Comment:
[Elsewhere we use 0 to denote 'not_tried'](https://github.com/code-dot-org/code-dot-org/blob/017b87c2cc38f698e334981096f2c54559f3892d/dashboard/app/helpers/application_helper.rb#L71-L72).